src/main/java/org/springframework/integration/aws/inbound/kinesis/KclMessageDrivenChannelAdapter.java

```diff
@@ -364,26 +365,34 @@ public Duration waitPeriodToDeleteFormerStreams() {
 
 				};
 
-		private final List<StreamConfig> streamConfigs =
-				Arrays.stream(KclMessageDrivenChannelAdapter.this.streams)
-						.map(streamName ->
-								new StreamConfig(StreamIdentifier.singleStreamInstance(streamName),
-										KclMessageDrivenChannelAdapter.this.streamInitialSequence))
-						.toList();
+		private List<StreamConfig> streamConfigs = new ArrayList<>();
 
 		StreamsTracker() {
 		}
 
+		private StreamConfig buildStreamConfig(String streamName) {
+			logger.info(LogMessage.format("Initializing stream config for Stream: [%s] for Multi Stream Tracker", streamName));
+
+			DescribeStreamSummaryResponse streamSummaryResponse = KclMessageDrivenChannelAdapter.this.kinesisClient.describeStreamSummary(DescribeStreamSummaryRequest.builder().streamName(streamName).build()).join();
+			return new StreamConfig(StreamIdentifier.multiStreamInstance(Arn.fromString(streamSummaryResponse.streamDescriptionSummary().streamARN()), streamSummaryResponse.streamDescriptionSummary().streamCreationTimestamp().toEpochMilli()),
```

Comment:
`toEpochSeconds()`.
I heard somewhere that it AWS services deal in seconds for timestamps.

The `streamDescriptionSummary()` can be extracted to variable.

We also prefer the length of the line not more than 120 symbols.